src/molecules/tanstack-table/tanstack-table-row-actions-confirmation.tsx

```diff
@@ -0,0 +1,77 @@
+'use client';
+
+import { Loader } from 'lucide-react';
+import * as React from 'react';
+
+import { Button } from '@/components/ui/button';
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@/components/ui/dialog';
+import { TanstackTableRowActionsConfirmationDialog } from './types';
+
+type TanstackTableConfirmationDialogProps<TData> = {
+  row: TData;
+  setDialogOpen: () => void;
+} & TanstackTableRowActionsConfirmationDialog<TData>;
+export function TanstackTableConfirmationDialog<TData>({
+  row,
+  title,
+  confirmationText,
+  onCancel,
+  onConfirm,
+  cancelText,
+  description,
+  setDialogOpen,
+}: TanstackTableConfirmationDialogProps<TData>) {
+  const dialogTitle = typeof title === 'function' ? title(row) : title;
+
+  const [isDeletePending, startDeleteTransition] = React.useTransition();
+
+  const handleOnConfirmClick = () => {
+    startDeleteTransition(() => {
+      onConfirm(row);
+      setDialogOpen();
+    });
+  };
+  const handleOnCancelClick = () => {
+    startDeleteTransition(() => {
+      onCancel(row);
+      setDialogOpen();
+    });
+  };
+
+  return (
+    <Dialog open onOpenChange={setDialogOpen}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>{dialogTitle}</DialogTitle>
+          <DialogDescription>{description}</DialogDescription>
+        </DialogHeader>
+        <DialogFooter className="gap-2 sm:space-x-0">
+          <DialogClose asChild>
+            <Button variant="outline" onClick={handleOnCancelClick}>
+              {cancelText}
+            </Button>
+          </DialogClose>
+          <Button
+            aria-label={confirmationText}
+            variant="destructive"
+            onClick={handleOnConfirmClick}
+            disabled={isDeletePending}
+          >
+            {isDeletePending && (
+              <Loader className="mr-2 size-4 animate-spin" aria-hidden="true" />
+            )}
+            {confirmationText}
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
```

Comment:
why did not we use our own dialog from the project?

Reply:
It's dialog from shadcn?

Reply:
we have this component: 
https://ayasofyazilim-clomerce.github.io/ayasofyazilim-ui/?path=/docs/molecules-dialog--docs

it would be nice to use it